UI/LinkText.cpp

```diff
@@ -92,6 +93,49 @@ namespace {
 
         return retval;
     }
+
+    const std::string FOCS_VALUE_TAG_OPEN_PRE("<" + VarText::FOCS_VALUE_TAG);
+    const std::string FOCS_VALUE_TAG_CLOSE("</" + VarText::FOCS_VALUE_TAG + ">");
+    const xpr::sregex FOCS_VALUE_SEARCH = FOCS_VALUE_TAG_OPEN_PRE >> xpr::_s >> (xpr::s1 = REGEX_NON_BRACKET) >> ">" >>
+                                          (xpr::s2 = REGEX_NON_BRACKET) >> FOCS_VALUE_TAG_CLOSE;
+
+    /** Parses VarText::FOCS_VALUE_TAG%s within @p text, replacing value ref name%s with
+     *  the evaluation result of that value ref.
+     *  Given tag content (i.e. the stringtable content for the tag name) gets added as explanation.
+     *  If the tag content is empty or @p use_description_instead_of_user_string is true,
+     *  the value ref description gets added as explanation instead. */
+    std::string ValueRefLinkText(const std::string& text, const bool use_description_instead_of_user_string) {
+        if (!boost::contains(text, FOCS_VALUE_TAG_CLOSE))
+            return text;
+
+        std::string retval(text);
+        auto text_it = retval.begin();
+        xpr::smatch match;
+
+        while (true) {
+            if (!xpr::regex_search(text_it, retval.end(), match, FOCS_VALUE_SEARCH, xpr::regex_constants::match_default))
+                break;
+
+            std::string value_ref_name(match[1]);
```

Comment:
maybe:

```
            std::string value_ref_name{match[1]};
            auto*       value_ref = GetValueRefBase(value_ref_name);
            auto        value_str{value_ref ? value_ref->StringResult() : value_ref_name};
            std::string explanation_str{
                use_description_instead_of_user_string && value_ref ?
                    (" (" + value_ref->Description() + ")") :
                    (match[2].length() == 0 ? "" : "( " + match[2] + ")")};
```

Reply:
not exactly what i wanted in UI, i will restructure like you suggest